app/test/search/tags_test.dart

```diff
@@ -139,5 +139,44 @@ void main() {
         ],
       });
     });
+
+    test('User-supplied queried tags #1', () async {
+      final index = SimplePackageIndex();
+      await index.addPackage(
+          PackageDocument(package: 'pkg1', tags: ['is:recent', 'is:legacy']));
+      await index
+          .addPackage(PackageDocument(package: 'pkg2', tags: ['is:legacy']));
+      await index.merge();
+
+      final rs =
+          await index.search(SearchQuery.parse(query: 'is:legacy -is:recent'));
+      expect(json.decode(json.encode(rs.toJson())), {
+        'indexUpdated': isNotEmpty,
+        'totalCount': 1,
+        'packages': [
+          {'package': 'pkg2', 'score': isNotNull},
+        ],
+      });
+    });
+
+    test('User-supplied queried tags #2', () async {
+      final index = SimplePackageIndex();
+      await index.addPackage(
+          PackageDocument(package: 'pkg1', tags: ['is:recent', 'is:legacy']));
+      await index
+          .addPackage(PackageDocument(package: 'pkg2', tags: ['is:recent']));
+      await index.merge();
+
+      final rs = await index.search(SearchQuery.parse(
+          tagsPredicate: TagsPredicate(prohibitedTags: ['is:legacy']),
```

Comment:
Should the `tagsPredicate` argument to `SearchQuery.parse` perhaps be named `defaultTagsPredicate`?